Source/CarthageKit/Project.swift

```diff
@@ -353,34 +353,71 @@ public final class Project {
+		return loadResolvedCartfile()
+			.flatMap(.Merge) {
+				return SignalProducer<Dependency<PinnedVersion>, CarthageError>(values: $0.dependencies)
+					.map { dependency in
+						return dependency.project
+					}
+			}
+			.flatMap(.Merge) { (projectIdentifier: ProjectIdentifier) -> SignalProducer<Dependency<SemanticVersion>, CarthageError> in
+				return self.versionsForProject(projectIdentifier)
+					.collect()
+					.map { (pinnedVersions: [PinnedVersion]) -> SemanticVersion? in
+						return pinnedVersions
+							.flatMap { SemanticVersion.fromPinnedVersion($0).value }
+							.maxElement()
+					}
+					.ignoreNil()
+					.map { Dependency(project: projectIdentifier, version: $0) }
+			}
+			.collect()
+	}
+	
 	/// Attempts to determine which of the project's Carthage
 	/// dependencies are out of date.
 	///
 	/// This will fetch dependency repositories as necessary, but will not check
 	/// them out into the project's working directory.
-	public func outdatedDependencies(includeNestedDependencies: Bool) -> SignalProducer<[(Dependency<PinnedVersion>, Dependency<PinnedVersion>)], CarthageError> {
+	public func outdatedDependencies(includeNestedDependencies: Bool) -> SignalProducer<[(Dependency<PinnedVersion>, Dependency<PinnedVersion>, Dependency<SemanticVersion>)], CarthageError> {
 		typealias PinnedDependency = Dependency<PinnedVersion>
-		typealias OutdatedDependency = (PinnedDependency, PinnedDependency)
+		typealias SemanticDependency = Dependency<SemanticVersion>
+		typealias OutdatedDependency = (PinnedDependency, PinnedDependency, SemanticDependency)
 
 		let currentDependencies = loadResolvedCartfile()
 			.map { $0.dependencies }
-		let updatedDependencies = updatedResolvedCartfile()
+		let satisfiableDependencies = updatedResolvedCartfile()
 			.map { $0.dependencies }
-		let outdatedDependencies = combineLatest(currentDependencies, updatedDependencies)
-			.map { (currentDependencies, updatedDependencies) -> [OutdatedDependency] in
+		
+		let outdatedDependencies = combineLatest(currentDependencies, satisfiableDependencies, latestDependencies())
+			.map { (currentDependencies, satisfiableDependencies, latestDependencies) -> [OutdatedDependency] in
 				var currentDependenciesDictionary = [ProjectIdentifier: PinnedDependency]()
 				for dependency in currentDependencies {
 					currentDependenciesDictionary[dependency.project] = dependency
 				}
-
-				return updatedDependencies.flatMap { updated -> OutdatedDependency? in
-					if let resolved = currentDependenciesDictionary[updated.project] where resolved.version != updated.version {
-						return (resolved, updated)
+				
+				var latestDependenciesDictionary = [ProjectIdentifier: SemanticDependency]()
+				for latestDependency in latestDependencies {
+					if let latestVersion = latestDependency.version.pinnedVersion, currentVersion = currentDependenciesDictionary[latestDependency.project] where currentVersion.version != latestVersion {
```

Comment:
Can you move this `where` clause to the `if` below inside `satisfiableDepndencies.flatMap`? I think that's a more readable approach.